substrate/thread_pool

```diff
@@ -99,23 +103,27 @@ namespace substrate
 		threadPool_t &operator =(const threadPool_t &) = delete;
 		threadPool_t &operator =(threadPool_t &&) = delete;
 
-		SUBSTRATE_NO_DISCARD(inline size_t numProcessors() const noexcept) { return affinity.numProcessors(); }
-		SUBSTRATE_NO_DISCARD(inline bool valid() const noexcept) { return !threads.empty(); }
-		SUBSTRATE_NO_DISCARD(inline bool ready() const noexcept) { return waitingThreads == affinity.numProcessors(); }
+		SUBSTRATE_NO_DISCARD(size_t numProcessors() const noexcept) { return affinity.numProcessors(); }
+		SUBSTRATE_NO_DISCARD(bool valid() const noexcept) { return !threads.empty(); }
+		SUBSTRATE_NO_DISCARD(bool ready() const noexcept) { return waitingThreads == affinity.numProcessors(); }
 
-		SUBSTRATE_NO_DISCARD(inline result_t queue(args_t ...args) noexcept)
+		SUBSTRATE_NO_DISCARD(result_t queue(args_t ...args) noexcept)
 		{
+			std::lock_guard<std::mutex> lock{workMutex};
```

Comment:
Does this not also need its own scope so it gets unlocked/discarded before calling `clearResultQueue()`?

Reply:
Maybe, the condition variable wakeup will cause all of them to wait until the queue is effectively released in the destructor scope.

I'll submit the change and see if CI agrees.